sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -144,9 +144,12 @@ case class GenerateExec(
       Seq.empty
     }
 
+    val requiredInput = input.zip(child.output).filter {
```

Comment:
```
    val requiredInput = if (requiredChildOutput.nonEmpty) {
        input.zip(child.output).filter {
        case(_, output) => parent.inputSet.contains(output)
      }.map(_._1)
    } else {
      Nil
    }
```
?

Reply:
Why do we need to look at `parent.input`? How about
```
val requiredAttrSet = AttributeSet(requiredChildOutput)
val requiredInput = child.output.zip(input).filter {
  case (attr, _) => requiredAttrSet.contains(attr)
}.map(_._2)
```